linux_sql/sql/queries.sql

```diff
@@ -1,20 +1,73 @@
 -- Group hosts by hardware info
-SELECT cpu_number, id AS host_id, total_mem
-FROM host_info
+SELECT 
+    cpu_number, 
+    id AS host_id,
+    total_mem
+FROM 
+    host_info
 ORDER BY
     cpu_number ASC,
     total_mem DESC
 ;
 
 -- Average memory usage 
-SELECT u.host_id, i.hostname, u.timestamp, 
+SELECT 
+    u.host_id,
+    i.hostname,
+    u.timestamp, 
     AVG(((i.total_mem - u.memory_free) / i.total_mem::float) * 100) OVER ( 
         PARTITION BY (
             DATE_TRUNC('hour', u.timestamp) + 
             DATE_PART('min', u.timestamp)::int / 5 * interval '5 min'
         )
     ) AS avg_used_mem_perc 
-FROM host_usage u 
-    INNER JOIN host_info i ON u.host_id=i.id
-ORDER BY u.host_id 
+FROM 
+    host_usage u INNER JOIN host_info i ON u.host_id = i.id
+ORDER BY
+    u.host_id 
 ; 
+
+-- Detect node failure
+/*
+CREATE TEMP TABLE test 
+( 
+    host_id             SMALLINT NOT NULL,
+    "timestamp"         TIMESTAMP NOT NULL,
+    is_failed           VARCHAR NOT NULL,
+    PRIMARY KEY("timestamp", host_id)
+);
+
+INSERT INTO test VALUES  
+    (1, '2020-03-11 10:23:01', 'ok'),
+    (1, '2020-03-11 10:24:01', 'failed'),
+    (1, '2020-03-11 10:25:01', 'failed'),
+    (1, '2020-03-11 10:26:01', 'failed'),
+    (1, '2020-03-11 10:27:01', 'ok')
+;
+*/
+
+WITH tmp AS (
```

Comment:
For this, I think the input will just be the host_usage table and we'll have to figure out how to find the missing timestamps for each host, count them to see if it's equal or greater than 3 and then list them

Reply:
I just made a release PR, will see.